lib/bas/shared_storage/postgres.rb

```diff
@@ -26,11 +26,16 @@ def read
       end
 
       def write(data)
-        params = { connection: write_options[:connection], query: write_query(data) }
+        return if write_options[:avoid_empty_data] && empty_data?(data)
```

Comment:
I think the empty_data should not be validated. The users should be able to avoid writing data in the shared storage even if there is data to save. I think this will help with flexibility for the user's use cases.